File: engine/engine-model/src/main/scala/tools/packet/guild/bbs/GetThreadList.scala
package tools.packet.guild.bbs

import client.database.data.BbsThreadData
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class GetThreadList(private var _threadData: java.util.List[BbsThreadData], private var _start: Int) extends PacketInput {
  def threadData: java.util.List[BbsThreadData] = _threadData

  def start: Int = _start

  override def opcode(): SendOpcode = SendOpcode.GUILD_BBS_PACKET
}

File: engine/engine-model/src/main/java/tools/packet/guild/bbs/GetThreadList.java
package tools.packet.guild.bbs;

import java.util.List;

import client.database.data.BbsThreadData;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record GetThreadList(List<BbsThreadData> threadData, Integer start) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.GUILD_BBS_PACKET;
   }
}